tools/ola_trigger/contrib/elgato_key_light.conf

```diff
@@ -0,0 +1,127 @@
+# OLA Trigger config to control Elgato Key Light or Key Light Airs
+# Copyright (C) 2020-2023 Johan Nilsson. https://gobo.ws
+
+
+
+# DMX chart
+
+###############################################################################
+
+# +---------+---------+----------------------------+
+# | Channel | Value   | Function                   |
+# +---------+---------+----------------------------+
+# | 1       | 0-100   | Dimmer (0-100%)            |
+# |         | 101-255 | Unused range               |
```

Comment:
> I think we should have the dimmer work over the full range rather than just part of it which will be rather confusing for users.
> 
> I suspect separately this means some sort of scale command would simplify repetitive config a lot.
> 
> I see @nomis52 did this originally with the Mac Volume example, I'm not sure I agree with it personally.

I totally agree with your opinion, but I don't see how it can be done in the configuration script since it does not support any scale commands.
Well the only thing to do it is to hard code the input range and the scaled range, something like this:

| Input range 0-255 | Output range 0-100 |
|-------------------|--------------------|
| 0                 | 0                  |
| 1                 | 0                  |
| 2                 | 1                  |
| 3                 | 1                  |
| 4                 | 2                  |
| 5                 | 2                  |
| 6                 | 2                  |
| 7                 | 3                  |
| 8                 | 3                  |
| 9                 | 4                  |
| 10                | 4                  |
| 11                | 4                  |
| 12                | 5                  |
| 13                | 5                  |
| 14                | 5                  |
| 15                | 6                  |
| 16                | 6                  |
| 17                | 7                  |
| 18                | 7                  |
| 19                | 7                  |
| 20                | 8                  |
| 21                | 8                  |
| 22                | 9                  |
| 23                | 9                  |
| 24                | 9                  |
| 25                | 10                 |
| 26                | 10                 |
| 27                | 11                 |
| 28                | 11                 |
| 29                | 11                 |
| 30                | 12                 |
| 31                | 12                 |
| 32                | 13                 |
| 33                | 13                 |
| 34                | 13                 |
| 35                | 14                 |
| 36                | 14                 |
| 37                | 15                 |
| 38                | 15                 |
| 39                | 15                 |
| 40                | 16                 |
| 41                | 16                 |
| 42                | 16                 |
| 43                | 17                 |
| 44                | 17                 |
| 45                | 18                 |
| 46                | 18                 |
| 47                | 18                 |
| 48                | 19                 |
| 49                | 19                 |
| 50                | 20                 |
| 51                | 20                 |
| 52                | 20                 |
| 53                | 21                 |
| 54                | 21                 |
| 55                | 22                 |
| 56                | 22                 |
| 57                | 22                 |
| 58                | 23                 |
| 59                | 23                 |
| 60                | 24                 |
| 61                | 24                 |
| 62                | 24                 |
| 63                | 25                 |
| 64                | 25                 |
| 65                | 25                 |
| 66                | 26                 |
| 67                | 26                 |
| 68                | 27                 |
| 69                | 27                 |
| 70                | 27                 |
| 71                | 28                 |
| 72                | 28                 |
| 73                | 29                 |
| 74                | 29                 |
| 75                | 29                 |
| 76                | 30                 |
| 77                | 30                 |
| 78                | 31                 |
| 79                | 31                 |
| 80                | 31                 |
| 81                | 32                 |
| 82                | 32                 |
| 83                | 33                 |
| 84                | 33                 |
| 85                | 33                 |
| 86                | 34                 |
| 87                | 34                 |
| 88                | 35                 |
| 89                | 35                 |
| 90                | 35                 |
| 91                | 36                 |
| 92                | 36                 |
| 93                | 36                 |
| 94                | 37                 |
| 95                | 37                 |
| 96                | 38                 |
| 97                | 38                 |
| 98                | 38                 |
| 99                | 39                 |
| 100               | 39                 |
| 101               | 40                 |
| 102               | 40                 |
| 103               | 40                 |
| 104               | 41                 |
| 105               | 41                 |
| 106               | 42                 |
| 107               | 42                 |
| 108               | 42                 |
| 109               | 43                 |
| 110               | 43                 |
| 111               | 44                 |
| 112               | 44                 |
| 113               | 44                 |
| 114               | 45                 |
| 115               | 45                 |
| 116               | 45                 |
| 117               | 46                 |
| 118               | 46                 |
| 119               | 47                 |
| 120               | 47                 |
| 121               | 47                 |
| 122               | 48                 |
| 123               | 48                 |
| 124               | 49                 |
| 125               | 49                 |
| 126               | 49                 |
| 127               | 50                 |
| 128               | 50                 |
| 129               | 51                 |
| 130               | 51                 |
| 131               | 51                 |
| 132               | 52                 |
| 133               | 52                 |
| 134               | 53                 |
| 135               | 53                 |
| 136               | 53                 |
| 137               | 54                 |
| 138               | 54                 |
| 139               | 55                 |
| 140               | 55                 |
| 141               | 55                 |
| 142               | 56                 |
| 143               | 56                 |
| 144               | 57                 |
| 145               | 57                 |
| 146               | 57                 |
| 147               | 58                 |
| 148               | 58                 |
| 149               | 59                 |
| 150               | 59                 |
| 151               | 59                 |
| 152               | 60                 |
| 153               | 60                 |
| 154               | 61                 |
| 155               | 61                 |
| 156               | 61                 |
| 157               | 62                 |
| 158               | 62                 |
| 159               | 63                 |
| 160               | 63                 |
| 161               | 63                 |
| 162               | 64                 |
| 163               | 64                 |
| 164               | 65                 |
| 165               | 65                 |
| 166               | 65                 |
| 167               | 66                 |
| 168               | 66                 |
| 169               | 67                 |
| 170               | 67                 |
| 171               | 67                 |
| 172               | 68                 |
| 173               | 68                 |
| 174               | 69                 |
| 175               | 69                 |
| 176               | 69                 |
| 177               | 70                 |
| 178               | 70                 |
| 179               | 71                 |
| 180               | 71                 |
| 181               | 71                 |
| 182               | 72                 |
| 183               | 72                 |
| 184               | 73                 |
| 185               | 73                 |
| 186               | 73                 |
| 187               | 74                 |
| 188               | 74                 |
| 189               | 75                 |
| 190               | 74                 |
| 191               | 75                 |
| 192               | 75                 |
| 193               | 76                 |
| 194               | 76                 |
| 195               | 76                 |
| 196               | 77                 |
| 197               | 77                 |
| 198               | 78                 |
| 199               | 78                 |
| 200               | 78                 |
| 201               | 79                 |
| 202               | 79                 |
| 203               | 80                 |
| 204               | 80                 |
| 205               | 80                 |
| 206               | 81                 |
| 207               | 81                 |
| 208               | 82                 |
| 209               | 82                 |
| 210               | 82                 |
| 211               | 83                 |
| 212               | 83                 |
| 213               | 84                 |
| 214               | 84                 |
| 215               | 84                 |
| 216               | 85                 |
| 217               | 85                 |
| 218               | 86                 |
| 219               | 86                 |
| 220               | 86                 |
| 221               | 87                 |
| 222               | 87                 |
| 223               | 88                 |
| 224               | 88                 |
| 225               | 88                 |
| 226               | 89                 |
| 227               | 89                 |
| 228               | 90                 |
| 229               | 90                 |
| 230               | 90                 |
| 231               | 91                 |
| 232               | 91                 |
| 233               | 92                 |
| 234               | 92                 |
| 235               | 92                 |
| 236               | 93                 |
| 237               | 93                 |
| 238               | 94                 |
| 239               | 94                 |
| 240               | 94                 |
| 241               | 95                 |
| 242               | 95                 |
| 243               | 96                 |
| 244               | 96                 |
| 245               | 96                 |
| 246               | 97                 |
| 247               | 97                 |
| 248               | 98                 |
| 249               | 98                 |
| 250               | 98                 |
| 251               | 99                 |
| 252               | 99                 |
| 253               | 99                 |
| 254               | 100                |
| 255               | 100                |


Otherwise this can be solved by adding a shell script but in this case I wanted to do a simple script without the need for any shell script.
For more advanced uses I use Node-RED and the Range node wich works fine for this kind of scaling.

Reply:
> I totally agree with your opinion, but I don't see how it can be done in the configuration script since it does not support any scale commands. Well the only thing to do it is to hard code the input range and the scaled range, something like this:

Yeah exactly, just like you're already doing with the colour temperature. Although as I mentioned with that if you map to a variable then you only need to write the actual command once. So it should only be 101-102 rows of config!

But yes, clearly a built in scale would be really helpful.

> Otherwise this can be solved by adding a shell script but in this case I wanted to do a simple script without the need for any shell script. For more advanced uses I use Node-RED and the Range node wich works fine for this kind of scaling.

Ah right. Is that the main missing feature from ola_trigger? Or are you generally doing more complicated logic when you're using Node Red?